departures.go

```diff
@@ -153,6 +171,50 @@ func main() {
 	}
 }
 
+func searchStations(name string) ([]station, error) {
+	var stations []station
+	err := getJSON(&stations, "https://2.bvg.transport.rest/locations?query=%s&poi=false&addresses=false", name)
+
+	return stations, err
+}
+
+func promptForStation(name string) (*station, error) {
+	stations, err := searchStations(name)
+	if err != nil {
+		return nil, fmt.Errorf("could not query stations")
+	}
+
+	if len(stations) == 0 {
+		// no stations found
+		return nil, fmt.Errorf("could not find matching stations")
+	}
+
+	// set first result as fallback
+	fallback := stations[0].Name
+
+	// convert to map[string]station to get station after user prompt
+	var options []string
+	optionStation := map[string]*station{}
+	for _, s := range stations {
+		options = append(options, s.Name)
+		optionStation[s.Name] = &s
+	}
```

Comment:
Is there maybe a way to get the index directly? This seems unnecessarily complex.

Reply:
The survey returns the exact string used for the options, so we'll need a way to map that the string back to the actual station. Maybe it can be a map of [name of station] to id. What do you think?

Reply:
> Maybe it can be a map of [name of station] to id. What do you think?

No, then leave it like this. Maybe we need the other data in the `station`.